### ISSUE TITLE
Is it possible to configure DBRider to fail a test if cleanup fails?

### ISSUE THREAD

**Issue by Livelypower** [database-rider/database-rider#225]
Is is possible to configure DBRider to hard fail a test if cleanBefore or cleanAfter could not be performed successfully?

From the docs I can see that DBR "tries" to clean the db using any supplied ordering and brute forcing. Is there any way to make a test fail if the database could not be cleaned?

**Comment by rmpestano** [comment]
Hi, I think if we throw an [exception here](https://github.com/database-rider/database-rider/blob/d33a38bf7589c45f905f92a8611e05ad940072a6/rider-core/src/main/java/com/github/database/rider/core/dataset/DataSetExecutorImpl.java#L624) instead of logging then it should fail.

But why do you need that? You have a passing test even when cleaning fails?

**Comment by Livelypower** [comment]
Yes, in some cases our current test suite will not be fully cleaned for the next test, which results in tests failing because of records already existing in our db.

**Comment by rmpestano** [comment]
Hi @Livelypower,

I've added a property called `raiseExceptionOnCleanUp` which defaults to false and you can enabled it in `@DBunit` or dbunit.yml (see [dbunit configuration](https://github.com/database-rider/database-rider#332-dbunit-configuration)).

You can try that with v`1.15.4-SNAPSHOT`, you'll need to [enable snapshots repo](https://github.com/database-rider/database-rider#snapshots) on your build to use that version.

Let me know if that works for you. Thanks for the feedback!

**Comment by Livelypower** [comment]
Hi @rmpestano 

I was able to use the snapshot version to try out `raiseExceptionOnCleanUp` to find tests that did not perform their cleanup correctly. Thanks a ton for your quick response!

We did notice however that once a test fails when the exception is thrown, there seems to be an issue with the connection to the db. We received timeouts from out connection pool. It wasn't an issue per se because the tests failed anyway, but maybe it's worth investigating.

Thanks again!
